app/actions/departments/department_test.go

```diff
@@ -0,0 +1,107 @@
+package departments_test
+
+import (
+	"mjm/app"
+	"mjm/app/models"
+	"net/http"
+	"testing"
+
+	"github.com/gobuffalo/suite/v4"
+	"github.com/wawandco/fako"
+)
+
+type ActionSuite struct {
+	*suite.Action
+}
+
+func Test_ActionSuite(t *testing.T) {
+	as := &ActionSuite{suite.NewAction(app.New())}
+	suite.Run(t, as)
+}
+
+func (as ActionSuite) Test_List() {
+	deparments := models.Department{}
+	fako.Fill(&deparments)
+
+	err := as.DB.Create(&deparments)
+	as.NoError(err)
+	res := as.HTML("/departments/list").Get()
+	body := res.Body.String()
+	as.Contains(body, deparments.Name)
+	as.Contains(body, deparments.Description)
+}
+
+func (as ActionSuite) Test_Create() {
+	deparments := &models.Department{}
+	fako.Fill(deparments)
+
+	res := as.HTML("/department/create").Post(deparments)
+
+	as.Equal(res.Code, http.StatusSeeOther)
+	as.Equal("/departments/list", res.Location())
+}
+
+func (as ActionSuite) Test_Update() {
+	deparments := &models.Department{}
+	fako.Fill(deparments)
+	err := as.DB.Create(deparments)
+	as.NoError(err)
+
+	departmentsUpdate := &models.Department{}
+	fako.Fill(departmentsUpdate)
+	departmentsUpdate.ID = deparments.ID
+
+	res := as.HTML("/update/%s", departmentsUpdate.ID).Put(departmentsUpdate)
+
+	as.Equal(res.Code, http.StatusSeeOther)
+	as.Equal("/departments/list", res.Location())
+	as.DB.Reload(deparments)
+	as.Equal(departmentsUpdate.Name, deparments.Name)
+}
```

Comment:
What are we updating here? And why you are checking whether the name is equal?

Reply:
Here what is being updated is the name that was generated by default with fako, since we indicated that it is going to generate two names at random and I am checking if it is the same name because when the Name field is updated it must have the same value in both departments. Name as in departmentUpdate.Name.

Reply:
This is a bit tricky, but I already understand what you did. Thanks